pkg/controller/operators/operatocondition_controller.go

```diff
@@ -0,0 +1,259 @@
+	for _, serviceAccount := range filterServiceAccounts(serviceAccountList.Items, hasOwnerFunc(ownerReference)) {
+		subjects = append(subjects, rbacv1.Subject{
+			Kind:     rbacv1.ServiceAccountKind,
+			Name:     serviceAccount.GetName(),
+			APIGroup: "",
+		})
+	}
+
+	role := rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            operatorCondition.GetName(),
+			Namespace:       operatorCondition.GetNamespace(),
+			OwnerReferences: []metav1.OwnerReference{ownerReference},
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:         []string{"get"},
+				APIGroups:     []string{"operators.coreos.com"},
+				Resources:     []string{"operatorconditions"},
+				ResourceNames: []string{operatorCondition.GetName()},
+			},
+			{
+				Verbs:         []string{"get,update,patch"},
+				APIGroups:     []string{"operators.coreos.com"},
+				Resources:     []string{"operatorconditions/status"},
+				ResourceNames: []string{operatorCondition.GetName()},
+			},
+		},
+	}
+	err = r.Client.Create(context.TODO(), &role)
+	if err != nil {
+		if !k8serrors.IsAlreadyExists(err) {
+
+			return err
+		}
+		existingRole := rbacv1.Role{}
+		err := r.Client.Get(context.TODO(), client.ObjectKey{Name: role.GetName(), Namespace: role.GetNamespace()}, &existingRole)
+		if err != nil {
+			return err
+		}
+		existingRole.OwnerReferences = role.OwnerReferences
+		existingRole.Rules = role.Rules
+		err = r.Client.Update(context.TODO(), &existingRole)
+		if err != nil {
+			return err
+		}
+	}
+
+	roleBinding := rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            operatorCondition.GetName(),
+			Namespace:       operatorCondition.GetNamespace(),
+			OwnerReferences: []metav1.OwnerReference{ownerReference},
+		},
+		Subjects: subjects,
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "Role",
+			Name:     role.GetName(),
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+	}
+	err = r.Client.Create(context.TODO(), &roleBinding)
+	if err != nil {
+		if !k8serrors.IsAlreadyExists(err) {
+			return err
+		}
+		existingRoleBinding := rbacv1.RoleBinding{}
+		err := r.Client.Get(context.TODO(), client.ObjectKey{Name: roleBinding.GetName(), Namespace: roleBinding.GetNamespace()}, &existingRoleBinding)
+		if err != nil {
+			return err
+		}
+		existingRoleBinding.OwnerReferences = roleBinding.OwnerReferences
+		existingRoleBinding.Subjects = roleBinding.Subjects
+		existingRoleBinding.RoleRef = roleBinding.RoleRef
+		err = r.Client.Update(context.TODO(), &existingRoleBinding)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func hasOwnerFunc(ownerReference metav1.OwnerReference) func(object metav1.Object) bool {
+	return func(object metav1.Object) bool {
+		if ownerutil.HasOwnerReference(object, ownerReference) {
+			return true
+		}
+		return false
+	}
+}
+
+func filterServiceAccounts(serviceAccounts []corev1.ServiceAccount, f func(object metav1.Object) bool) []corev1.ServiceAccount {
+	// // filter based on OwnerReferences
```

Comment:
Very consequential suggestion:
```suggestion
	// filter based on OwnerReferences
```

Reply:
Addressed in the latest version of the PR.